src/Transport/Config/DefaultConfigurationValues.cs

```diff
@@ -48,7 +48,7 @@ void ApplyDefaultValuesForAddressing(SettingsHolder settings)
 
         void ApplyDefaultsForConnectivity(SettingsHolder settings)
         {
-            var numberOfLogicalCores = Math.Max(0, Environment.ProcessorCount);
+            var numberOfLogicalCores = Environment.ProcessorCount;
```

Comment:
We probably want to do something like Core does: `Max(2, Environment.ProcessorCount)`.